### DISCUSSION TITLE
Preinstalled softwares

### DISCUSSION THREAD

**Discussion by clemlesne** [clemlesne/blue-agent#288] [Q&A] [answered]
Do you need different, more or less pre installed softwares?

**Comment by lucsoft** [comment]
Well i would love if packages for playwright would be pre-installed as an opt in

**Reply by clemlesne** [reply]
Understood, the issue with playwright is the weight. It approximates 1GB with Chromium, FFPEG and system dependencies…

Another approach would be to use the pre-built official container (https://hub.docker.com/r/microsoft/playwright) specifically to run your tests, but it requires a bit more privileges in Kubernetes (see docker in docker section in the doc). Would it fit your need?

Finally, don’t hesitate for that kind of specific usage, to build your special container on top of this project. Then you can deploy it with a demand (see the doc for the demands configuration).

**Reply by lucsoft** [reply]
Image size is not really a problem (because caching)

i would like to avoid Docker in Docker, i don't want to add too much complexity to my CI runs

well yeah i could but then i also need to care about updating and i need to setup some CI for it.
But i think the last one is currently my only good option (but having first class support would be better)

**Reply by clemlesne** [reply, marked as answer]
To better understand, how much times does your install step take?

Are you using all-in-one step from playwright CLI `npx playwright install --with-deps chromium`?

Plus, are you [caching playwright cache folder](https://playwright.dev/docs/browsers#managing-browser-binaries)? See [the cache step](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/cache-v2?view=azure-pipelines).

**Reply by clemlesne** [reply]
> but having first class support would be better

It is good to remind that this is a third-party project supported by a single individual (me). I'm just a geek willing to help the community with a side project :)

[Managed DevOps Pools](https://learn.microsoft.com/en-us/azure/devops/managed-devops-pools/?view=azure-devops) is the official implementation which looks the most similar to Blue Agent.
